### PR TITLE
Prohibit using FreeObject as a base or linking to it

### PR DESCRIPTION
**Pull request by elprans**
Neither makes sense semantically.

### CODE REVIEW

File: edb/schema/indexes.py [hunk 1]
```diff
@@ -222,6 +225,7 @@ def get_object(  # NoQA: F811
         *,
         name: Optional[sn.Name] = None,
         default: Union[Index, so.NoDefaultT, None] = so.NoDefault,
+        sourcectx: Optional[parsing.ParserContext] = None,
```
Comment (1st1) [inline edb/schema/indexes.py:228]: This new param isn't used in the body of the function

File: edb/schema/objtypes.py [hunk 2]
```diff
@@ -100,10 +100,15 @@ def is_object_type(self) -> bool:
...
 
     def is_free_object_type(self, schema: s_schema.Schema) -> bool:
-        return self.issubclass(
-            schema,
-            schema.get('std::FreeObject', type=ObjectType),
-        )
+        if self.get_name(schema) == sn.QualName('std', 'FreeObject'):
+            return True
+
+        FreeObj = schema.get('std::FreeObject', type=ObjectType, default=None)
```
Comment (1st1) [inline edb/schema/objtypes.py:106]: nit: I'd keep the full name ;)

```suggestion
        FreeObject = schema.get('std::FreeObject', type=ObjectType, default=None)
```

File: tests/test_schema.py [hunk 3]
```diff
@@ -275,26 +275,44 @@ def test_schema_overloaded_link_04(self):
...
 
+    @tb.must_fail(
+        errors.InvalidLinkTargetError,
+        "object type 'std::FreeObject' is not a valid link target",
+        position=69,
+    )
+    def test_schema_bad_link_03(self):
+        """
+            type Object {
+                link foo -> FreeObject
```
Comment (1st1) [inline tests/test_schema.py:312]: Can you add one more test testing `link foo := (SELECT {a := 1})`?

**Review by 1st1 — APPROVED** [review]
I've left a couple nit comments. I'm not sure if what's in this PR is enough to prohibit computed links with expression returning a FreeObject; if it is, it needs a test. If not, a follow up PR?

**Comment by elprans** [comment]
> f it is, it needs a test. If not, a follow up PR?

A follow up, because the fix is not straightforward.

**Pull request merged**